File: backend/api/controllers/user_controller.py
from rest_framework.decorators import api_view
from rest_framework.request import Request

from api.middleware.auth_middleware import get_user_from_authorization_header
from api.models import Usuario
from api.utils import response_helper


@api_view(["GET"])
def get_me(request: Request):
    decoded = get_user_from_authorization_header(request.headers.get("Authorization"))
    user_id = decoded.get("id")
    
    # Buscar dados completos do usuário
    try:
        user = Usuario.objects.get(idusuario=user_id)
    except Usuario.DoesNotExist:
        return response_helper.error("Usuário não encontrado", status_code=404)
    
    # Retornar dados sem informações sensíveis (igual ao Node)
    user_dict = user.to_dict(exclude_sensitive=True)
    
    return response_helper.success(user_dict, "Dados do usuário")

File: back-end/src/controllers/user.controller.js
const usuarioModel = require('../models/Usuario');
const responseHelper = require('../utils/responseHelper');
const asyncHandler = require('../utils/asyncHandler');

const userController = {
  /**
   * Retorna dados do usuário autenticado
   */
  getMe: asyncHandler(async (req, res) => {
    // req.user é definido pelo middleware authenticate
    const userId = req.userId;
    
    // Buscar dados completos do usuário
    const user = await usuarioModel.findById(userId);
    
    if (!user) {
      return responseHelper.error(res, 'Usuário não encontrado', 404);
    }

    // Retornar dados sem informações sensíveis
    const { senha, cpf, numero_telefone, ...userWithoutSensitiveData } = user;
    
    return responseHelper.success(res, userWithoutSensitiveData, 'Dados do usuário');
  }),
};

module.exports = userController;



